File: java/src/main/java/com/workshop/architecture/fitness/application/port/inbound/ActivateMembership.java
package com.workshop.architecture.fitness.application.port.inbound;

import com.workshop.architecture.fitness.application.ActivateMembershipInput;
import com.workshop.architecture.fitness.application.ActivateMembershipResult;
import jakarta.transaction.Transactional;

public interface ActivateMembership {
    @Transactional
    ActivateMembershipResult activateMembership(ActivateMembershipInput input);
}

File: python/src/workshop_api/fitness/hexagon/inside/port/inbound/activate_membership.py
from typing import Protocol

from workshop_api.fitness.hexagon.inside.port.inbound.activate_membership_input import (
    ActivateMembershipInput,
)
from workshop_api.fitness.hexagon.inside.port.inbound.activate_membership_result import (
    ActivateMembershipResult,
)


class ForActivatingMemberships(Protocol):
    async def activate_membership(
        self,
        input_data: ActivateMembershipInput,
    ) -> ActivateMembershipResult:
        pass
